hera_mc/cm_health.py

```diff
@@ -59,39 +67,37 @@ def ensure_conndict(self):
             k = ':'.join(connection)
             if k in self.conndict.keys():
                 self.multiples.add(k)  # only add to multiples if there is more than one.
-            self.conndict.setdefault(k, []).append([cm_utils.get_astropytime(conn.start_gpstime),
-                                                    cm_utils.get_stopdate(conn.stop_gpstime)])
+            self.conndict.setdefault(k, []).append(conn)
 
     def check_for_duplicate_connections(self):
         """
         Checks all of the self.multiple keys to see if any of them overlap in time.
         If they do, it is a conflicting duplicate connection.
-        Writes a class variable:
-        duplicates - list - [[key, i, j], ...]
-            List which keeps triplets of duplication information as the key and the
-            indices of the conflicting times.
 
         Returns the number of duplicates found.
         """
         self.ensure_conndict()
-        if len(self.multiples) == 0:
-            print("No duplications found.")
-            return 0
         self.duplicates = []
         for k in self.multiples:
             for i in range(len(self.conndict[k])):
                 for j in range(i):
-                    if check_for_overlap(self.conndict[k][i], self.conndict[k][j]):
+                    intervals = [[self.conndict[k][i].start_gpstime, self.conndict[k][i].stop_gpstime],
+                                 [self.conndict[k][j].start_gpstime, self.conndict[k][j].stop_gpstime]]
+                    if check_for_overlap(intervals):
                         self.duplicates.append([k, i, j])
         if len(self.duplicates):
-            print('{} duplications found.'.format(len(self.duplicates)))
-            for d in self.duplicates:
-                i = self.conndict[d[0]][d[1]]
-                j = self.conndict[d[0]][d[2]]
-                print('\t{} <1>{}-{}  <2>{}-{}'.format(k, i[0], i[1], j[0], j[1]))
+            s = 's'
+            if len(self.duplicates) == 1:
+                s = ''
+            print('{} duplicate{} found.'.format(len(self.duplicates), s))
+            for dup in self.duplicates:
+                dc = self.conndict[dup[0]]
+                i0 = [dc[dup[1]].start_gpstime, dc[dup[1]].stop_gpstime]
+                i1 = [dc[dup[2]].start_gpstime, dc[dup[2]].stop_gpstime]
+                print('\t{}  --->  {} {}'.format(dc[0], i0, i1))
         else:
             print('No duplications found.')
-        print('Out of {} connections checked.'.format(self.num_connections))
+        print('{} connections checked.'.format(self.num_connections))
```

Comment:
Should it always print these messages, even if it's being called programmatically?